ios/Classes/AudioServicePlugin.m

```diff
@@ -376,7 +376,7 @@ - (void) updateControl:(enum MediaAction)action {
     MPRemoteCommand *command = commands[action];
     if (command == [NSNull null]) return;
     int actionBit = 1 << action;
-    BOOL enable = actionBits & actionBit;
+    BOOL enable = ((actionBits >> action) & 1);
```

Comment:
Nice find!

The code's a bit subtle so it might require a comment, otherwise I wonder if this would be more intuitive:

```objc
BOOL enable = (actionBits & actionBit) != 0;
```

(Haven't tested at all).

Reply:
I'd prefer to go with a comment, as it seems cleaner to me - it's just a bit shift, rather than a bit shift and a comparison.

Reply:
Given I haven't tested my code, let's go with yours ;-)

I think `(bits & bit)` is a more common way of testing for a bit, so I was just trying to think of a way to get that to work with assignment conversion to BOOL. Probably a simpler way would actually be to change `BOOL` to `long` to match the type of `actionBits` but again I'm happy to go with your code since it's been tested.

Reply:
I know it hardly matters, but using a long would use more memory unneccesarily. If you're okay with the latest code, I'll mark this as resolved, I guess.